### ISSUE TITLE
Product duplication does not copy 'When out of stock' value correctly

### ISSUE THREAD

**Issue by therampagerado** [thirtybees/thirtybees#1756]
Current system: edge.

Multistore: enabled

Currently the system does not duplicate the status of 'When out of stock'.

![image](https://github.com/thirtybees/thirtybees/assets/11251370/9f448cc1-021a-4baf-9ee7-5870f597435d)

**Comment by getdatakick** [comment]
AdminProductsControllerCore::processDuplicate should duplicate StockAvailable outOfStock settings (for product and all combinations)

**Comment by therampagerado** [comment]
So you're unable to replicate it?

The scenario is:

I have a product with Allow orders.
I duplicate it.
The duplicate product have all other settings duplicated but 'When out of stock' is always 'Default: Deny orders' (in my 2 instances both running edge - the default is Deny orders.

**Comment by getdatakick** [comment]
No, my comment was just to pinpoint the location, and what to fix.
